flask-version/metagenomongo/module/validation.py

```diff
@@ -0,0 +1,125 @@
+            return False
+
+def create_data_type_list(data_type, fields, options):
+    data_list = []
+    for field in fields:
+        
+        if data_type == options[field]['datatype']:
+           data_list.append(field)
+        elif data_type == options[field]['datatype']:
+            data_list.append(field)
+    return data_list
+
+def validation_all(expected_headers, fields, options, results, df_temp):
+    df_temp = df_temp.reindex(columns=expected_headers, fill_value='')  # Ensure columns match the expected headers
+    df_temp = df_temp.loc[:, ~df_temp.columns.duplicated()]  # Remove any duplicate columns              
+    
+    # Remove fully empty rows
+    df_temp = df_temp[~(df_temp == '').all(axis=1)]
+    data_tolist = df_temp.values.tolist()
+    data = df_temp
+    corrected_count = 0
+    corrected_items = []
+    invalid_date_messages = []
+    invalid_combobox_messages = []
+    data = df_temp.values.tolist()
+    int_dynamic_type = create_data_type_list("int", fields, options)
+    float_dynamic_type = create_data_type_list("float", fields, options)
+    # Apply corrections to data
+    for row_index, row in enumerate(data):
+        for col_index, cell in enumerate(row):
+            if isinstance(cell, str):  # Process only strings
+                original_cell = cell
+                cell = cell.strip(',; ')
+                cell = re.sub(r'\s+', ' ', cell)
+                cell = re.sub(r',\s*(\S)', r'; \1', cell)
+                
+                # Apply specific column corrections
+                if fields[col_index] == "locality":
+                    cell = re.sub(r':(?!\s)', ': ', cell)
+                elif fields[col_index] == "source_type":
+                    cell = cell.capitalize()
+                elif fields[col_index] == "if_repeated":
+                    cell = "y" if cell.lower() == "yes" else "n" if cell.lower() == "no" else cell
+                
+                if cell != original_cell:
+                    data[row_index][col_index] = cell
+                    corrected_count += 1
+                    corrected_items.append(f'Row {row_index + 1}, Column {col_index + 1}: {original_cell} -> {cell}')
+            
+            elif isinstance(cell, datetime.datetime):  # Handle datetime objects separately
+                original_cell = cell
+                cell = cell.strftime('%Y-%m-%d %H:%M:%S')
+                if cell != original_cell:
+                    data[row_index][col_index] = cell
+                    corrected_count += 1
+                    corrected_items.append(f'Row {row_index + 1}, Column {col_index + 1}: {original_cell} -> {cell}')
+    date_pattern = re.compile(r'^\d{4}-\d{2}-\d{2}(T\d{2}:\d{2}:\d{2}\.\d{3}Z)?$') 
+    # Validate data
+    for row_index, row in enumerate(data):
+        for col_index, cell in enumerate(row):
+            header = fields[col_index]
+            if isinstance(cell, str):
+                if header in ["collection_date", "run_date"]:
```

Comment:
A list like this is problematic for maintenance. Introduce a module-level constant for it like
```
DATE_FIELDS = ["collection_date", "run_date"]
```